packages/appkit/src/client.ts

```diff
@@ -2165,4 +2172,61 @@ export class AppKit {
 
     return this.initPromise
   }
+
+  private async checkExistingConnection() {
+    try {
+      if (!CoreHelperUtil.isTelegram()) {
+        return
+      }
+      const socialProviderToConnect = SafeLocalStorage.getItem(
```

Comment:
I think we have this in `StorageUtil` we are centralizing these storage behaviors there so we can more easily migrate to SSR handling 🙏